### PR TITLE
chore: Remove DepositTracker from the IngressEgress pallet

### PR DESCRIPTION
**Pull request by syan095**
# Pull Request

Closes: PRO-1485

## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [x] I have updated documentation where appropriate.

## Summary

Removed the redundant Deposit Tracker from the ingress-egress pallet.

### CODE REVIEW

File: state-chain/pallets/cf-ingress-egress/src/lib.rs [hunk 1]
```diff
@@ -545,10 +495,6 @@ pub mod pallet {
 	pub type FailedForeignChainCalls<T: Config<I>, I: 'static = ()> =
 		StorageMap<_, Twox64Concat, EpochIndex, Vec<FailedForeignChainCall>, ValueQuery>;
 
-	#[pallet::storage]
-	pub type DepositBalances<T: Config<I>, I: 'static = ()> =
```
Comment (kylezs) [inline state-chain/pallets/cf-ingress-egress/src/lib.rs:549]: this needs a migration to remove the storage

**Comment by dandanlen** [comment]
Not sure what's up with CI, will take a look next week.

**Comment by syan095** [comment]
Merged from main - fixed CI problem.
Also added a comment to the 11->12 migration as it is a "placeholder" for a migration written in the Release branch only.

**Comment by kylezs** [comment]
Just resolved the conflicts with #5105 , should be ready to go @dandanlen

File: state-chain/pallets/cf-ingress-egress/src/migrations.rs [hunk 2]
```diff
@@ -1,4 +1,8 @@
 use crate::Pallet;
-use cf_runtime_upgrade_utilities::PlaceholderMigration;
+use cf_runtime_upgrade_utilities::{PlaceholderMigration, VersionedMigration};
+mod remove_deposit_tracker;
 
-pub type PalletMigration<T, I> = PlaceholderMigration<Pallet<T, I>, 12>;
+pub type PalletMigration<T, I> = (
+	VersionedMigration<Pallet<T, I>, remove_deposit_tracker::Migration<T, I>, 12, 13>,
+	PlaceholderMigration<Pallet<T, I>, 13>,
+);
```
Comment (dandanlen) [inline state-chain/pallets/cf-ingress-egress/src/migrations.rs:8]: Looks like something might have got lost/mixed up in a merge - pallet version is still 12.

**Comment by dandanlen** [comment]
@syan095 can you look into what is going wrong with this?

**Comment by kylezs** [comment]
I think it just needs a rebase, didn't get round to it - there were some solana tests using external RPCs (which I think broke overnight), I have since disabled those tests on main, so a rebase/merge with main should help 🤞

**Review by kylezs — APPROVED** [review]

**Pull request merged**